Bad version in the dev branch

It looks like our uploader did not like the version used for the dev branch. I'm investigating locally what we can do about it.

xref #36

The problem is that the version is blank when uploading and it looks for `pynio--py27_0.tar.bz2` instead of `pynio-1.5.1.dev0+05806a2-py27_0.tar.bz2`.

@msarahan any advice on how to write version like that? They are pep440 valid according to [this](https://stackoverflow.com/questions/35522047/pep440-compliant-and-git-describe-info-available-from-deployed-package), but maybe that is a corner case...

```
TEST END: /feedstock_root/build_artefacts/linux-64/pynio-1.5.1.dev0+05806a2-py27_0.tar.bz2
INFO:conda_build.config:--dirty flag and --keep-old-work not specified.Removing build/test folder after successful build/test.

# Automatic uploading is disabled
# If you want to upload package(s) to anaconda.org later, type:

anaconda upload /feedstock_root/build_artefacts/linux-64/pynio-1.5.1.dev0+05806a2-py27_0.tar.bz2

# To have conda build upload to anaconda.org automatically, use
# $ conda config --set anaconda_upload yes

anaconda_upload is not set.  Not uploading wheels: []
+ upload_or_check_non_existence /recipe_root conda-forge --channel=dev
Using Anaconda API: https://api.anaconda.org
Processing pynio
Using Anaconda API: https://api.anaconda.org
[ERROR] file /feedstock_root/build_artefacts/linux-64/pynio--py27_0.tar.bz2 does not exist
Traceback (most recent call last):
  File "/opt/conda/bin/upload_or_check_non_existence", line 135, in <module>
    main()
  File "/opt/conda/bin/upload_or_check_non_existence", line 124, in main
    upload(cli, meta, owner, channel)
  File "/opt/conda/bin/upload_or_check_non_existence", line 75, in upload
    env=os.environ)
  File "/opt/conda/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['anaconda', '--quiet', '-t', '/tmp/tmp02ergm_k/binstar.token', 'upload', '/feedstock_root/build_artefacts/linux-64/pynio--py27_0.tar.bz2', '--user=conda-forge', '--channel=dev']' returned non-zero exit status 1.
```

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -1,4 +1,4 @@
-{% set version = "1.5.1.dev0+" + GIT_FULL_HASH[0:7] %}
+{% set version = "1.5.1.dev0" %}
 {% set commit_id = "05806a26a8676fea5832c0f270609e20df1d7276" %}
 
 package:

